### PR TITLE
adopt pure renderer/client-side ui

### PR DESCRIPTION
**Pull request by connor4312**
Looks and functions just like the extension-host-rendered HTML did. I used preact in the renderer since it's a nice and very small library. 

As referenced in https://github.com/microsoft/vscode/issues/102644, unfortunately I had to make two webpack builds since, although webpack supports multi-entrypoint builds, you can't change the `libraryType` between them; the renderer needs to be IIFE.

### CODE REVIEW

File: src/renderer/rendereApi.d.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Type definitions for non-npm package vscode-notebook-renderer 1.47
```
Comment (connor4312) [inline src/renderer/rendereApi.d.ts:1]: Can be removed once https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46331 is merged

**Comment by jrieken** [comment]
> ... I had to make two webpack builds since, although webpack supports multi-entrypoint builds, you can't change the libraryType between them; the renderer needs to be IIFE.

I think that's good because it adds more clarity and brings it in sync with the tsconfig-files. I like that outcome. 

> I used preact in the renderer since it's a nice and very small library.

What happened to good old timeless html...

File: src/renderer/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "lib": ["ES2019", "dom"],
+    "jsx": "react",
+    "jsxFactory": "h",
+  },
+  "references": [
+    {
+      "path": "../common"
+    }
+  ]
+}
```
Comment (jrieken) [inline src/renderer/tsconfig.json:14]: I like this but there is also `tsconfig.renderer.json`. WHich one is obsolete?

Comment (connor4312) [reply src/renderer/tsconfig.json:14]: tsconfig.renderer.json is no longer used I believe

Comment (jrieken) [reply src/renderer/tsconfig.json:14]: alright. I nuke it

**Pull request merged**